### PR TITLE
Fix edge cases in creating and updating Hyperdrive over Access configs

### PR DESCRIPTION
**Pull request by OilyLime**
## What this PR solves / how to test

Fixes [SQC-308](https://jira.cfdata.org/browse/SQC-308) [SQC-309](https://jira.cfdata.org/browse/SQC-309)

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Tests included
  - [ ] Tests not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because: Wrangler Hyperdrive commands not in E2E CI Jobs
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Changeset included
  - [ ] Changeset not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: Doesn't change documented wrangler commands

### CODE REVIEW

File: packages/wrangler/src/hyperdrive/update.ts [hunk 1]
```diff
@@ -94,23 +106,12 @@ export async function handler(
...
-	const noOriginFieldSet = requiredOriginOptions.every(
-		(field) => !isOptionSet(args, field)
-	);
-
-	if (!allOriginFieldsSet && !noOriginFieldSet) {
-		throw new UserError(
-			`When updating the origin, all of the following must be set: ${requiredOriginOptions
-				.map((option) => camelToKebab(option))
-				.join(", ")}`
-		);
```
Comment (penalosa) [inline packages/wrangler/src/hyperdrive/update.ts:109]: Why is this being removed? It's a much more user friendly error than the default yargs one:

```
Implications failed:
   database -> origin-host database -> origin-user database -> origin-password
```

File: packages/wrangler/src/hyperdrive/create.ts [hunk 2]
```diff
@@ -145,9 +144,9 @@ export async function handler(
 		const origin =
 			args.accessClientId && args.accessClientSecret
 				? {
-						host: url.hostname,
+						host: url.hostname + url.pathname,
```
Comment (penalosa) [inline packages/wrangler/src/hyperdrive/create.ts:147]: This doesn't seem right? Why is the pathname being passed to `host`?

File: packages/wrangler/src/hyperdrive/create.ts [hunk 3]
```diff
@@ -157,7 +156,7 @@ export async function handler(
 						host: url.hostname,
 						port: parseInt(url.port),
 						scheme: url.protocol.replace(":", ""),
-						database: decodeURIComponent(url.pathname.replace("/", "")),
+						database: args.connectionString !== "" ? decodeURIComponent(url.pathname.replace("/", "")) : args.database,
```
Comment (penalosa) [inline packages/wrangler/src/hyperdrive/create.ts:159]: That's kinda weird escaping behaviour—is this documented somewhere?

File: .changeset/ninety-icons-live.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+Fix rough edges when creating and updating Hyperdrive over Access configs
```
Comment (edmundhung) [inline .changeset/ninety-icons-live.md:5]:
```suggestion
fix: rough edges when creating and updating Hyperdrive over Access configs
```

File: packages/wrangler/src/hyperdrive/update.ts (hunk 1 above)
Comment (OilyLime) [reply packages/wrangler/src/hyperdrive/update.ts:109]: I prefer to let yargs handle that instead of writing my own logic and risk introducing bugs.

File: packages/wrangler/src/hyperdrive/create.ts (hunk 2 above)
Comment (OilyLime) [reply packages/wrangler/src/hyperdrive/create.ts:147]: It's a special case, we allow paths on Hyperdrive over Access hostnames, since they represent Cloudflare Tunnel endpoints which can have paths.

**Review by edmundhung — APPROVED** [review]
